framework/src/constraints/ADMortarConstraint.C

```diff
@@ -199,40 +199,27 @@ ADMortarConstraint::trimDerivative(const dof_id_type & remove_derivative_index,
 void
 ADMortarConstraint::trimInteriorNodeDerivatives(
     const std::map<unsigned int, unsigned int> & domain_ip_lowerd_map,
-    std::array<MooseVariable *, 3> & moose_var,
-    ADReal & var1,
-    ADReal & var2,
-    ADReal & var3,
+    const std::vector<const MooseVariable *> & moose_vars,
+    std::vector<ADReal *> & ad_vars,
     const bool is_secondary)
 {
   // Remove interior node variable's derivatives from AD objects.
   for (const auto dof_index :
        (is_secondary ? make_range(_test_secondary.size()) : make_range(_test_primary.size())))
     if (!domain_ip_lowerd_map.count(dof_index))
     {
-      if (moose_var[0])
-      {
-        const auto & remove_derivative_index_x =
-            is_secondary ? moose_var[0]->dofIndices()[dof_index]
-                         : moose_var[0]->dofIndicesNeighbor()[dof_index];
-        trimDerivative(remove_derivative_index_x, var1);
-      }
-
-      if (moose_var[1])
-      {
-        const auto & remove_derivative_index_y =
-            is_secondary ? moose_var[1]->dofIndices()[dof_index]
-                         : moose_var[1]->dofIndicesNeighbor()[dof_index];
-        trimDerivative(remove_derivative_index_y, var2);
-      }
-
-      if (moose_var[2])
-      {
-        const auto & remove_derivative_index_z =
-            is_secondary ? moose_var[2]->dofIndices()[dof_index]
-                         : moose_var[2]->dofIndicesNeighbor()[dof_index];
-        trimDerivative(remove_derivative_index_z, var3);
-      }
+      for (const auto moose_var_index : make_range(moose_vars.size()))
+        if (moose_vars[moose_var_index])
```

Comment:
are there cases where we put `nullptrs` in this container?

Reply:
Here, for example, https://github.com/idaholab/moose/blob/239e52b718b348acd7376dfa8f9db26c7538ed81/modules/contact/src/constraints/ComputeDynamicWeightedGapLMMechanicalContact.C#L91. But I could do `push_back` operations in the contact constraints instead and avoid it.

Reply:
With the latest push we shouldn't need to check for `nullptr` anymore.